tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -111,25 +111,6 @@ static PyObject * THPVariable_dim(PyObject* self, PyObject* args)
    END_HANDLE_TH_ERRORS
 }
 
-static Tensor dispatch_contiguous(const Tensor & self) {
-  AutoNoGIL no_gil;
-  DeviceGuard device_guard(self);
-  return self.contiguous();
-}
-
-static PyObject * THPVariable_contiguous(PyObject* self, PyObject* args)
-{
-  HANDLE_TH_ERRORS
-  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
-  // avoids touching the GIL or current device if self is already contiguous
```

Comment:
I think it does touch the GIL/device, but i'm not sure how to get tracing working properly along with this short-circuiting behavior. Any ideas?

Reply:
Yeah.. this seems hard without some special casing. Lose some gain some. I think this change is fine, although it would be interesting to see how much time skipping `AutoNoGIL` and `DeviceGuard` actually saves.

Reply:
FWIW, here are the results:
```
In [3]: x = torch.randn(200, 200)

# Before patch
In [4]: %timeit x.contiguous()
164 ns ± 3.44 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

# After patch
In [3]: %timeit x.contiguous()
425 ns ± 4.86 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
We have a way to make sure we don't release GIL, see `with_gil`. But it might not be supported on `native_functions.yaml` at the moment.